components/Tabs/examples/supplemental/TabBarIconExampleSupplemental.m

```diff
@@ -22,33 +22,215 @@
+
+  self.alignmentButton.translatesAutoresizingMaskIntoConstraints = NO;
+  [NSLayoutConstraint constraintWithItem:self.alignmentButton
+                               attribute:NSLayoutAttributeCenterX
+                               relatedBy:NSLayoutRelationEqual
+                                  toItem:self.view
+                               attribute:NSLayoutAttributeCenterX
+                              multiplier:1
+                                constant:0]
+      .active = YES;
+  [NSLayoutConstraint constraintWithItem:self.alignmentButton
+                               attribute:NSLayoutAttributeBottom
+                               relatedBy:NSLayoutRelationEqual
+                                  toItem:self.view
+                               attribute:NSLayoutAttributeBottom
+                              multiplier:1.0
+                                constant:-40]
+      .active = YES;
+
+  [self.alignmentButton setTitle:@"Change Alignment" forState:UIControlStateNormal];
+  [self.alignmentButton addTarget:self
+                           action:@selector(changeAlignment:)
+                 forControlEvents:UIControlEventTouchUpInside];
+}
+
+- (void)setupAppBar {
   self.view.backgroundColor = [UIColor whiteColor];
 
+  self.appBar = [[MDCAppBar alloc] init];
+  [self addChildViewController:self.appBar.headerViewController];
+
+  self.appBar.headerViewController.headerView.backgroundColor = [UIColor whiteColor];
+  self.appBar.headerViewController.headerView.minimumHeight = 76 + 72;
+  self.appBar.headerViewController.headerView.tintColor = [[MDCPalette bluePalette] tint500];
+
+  [self.appBar addSubviewsToParent];
+
   UIBarButtonItem *badgeIncrementItem =
-      [[UIBarButtonItem alloc] initWithTitle:@"Increment"
+      [[UIBarButtonItem alloc] initWithTitle:@"Add"
                                        style:UIBarButtonItemStylePlain
                                       target:self
                                       action:@selector(incrementBadges:)];
   self.navigationItem.rightBarButtonItem = badgeIncrementItem;
 
-  // Button to change tab alignments.
-  self.alignmentButton = [[MDCRaisedButton alloc] init];
-  [self.alignmentButton setTitle:@"Change Alignment" forState:UIControlStateNormal];
-  [self.alignmentButton sizeToFit];
-  self.alignmentButton.center = CGPointMake(CGRectGetMidX(self.view.bounds), 100);
-  self.alignmentButton.autoresizingMask = UIViewAutoresizingFlexibleLeftMargin |
-                                          UIViewAutoresizingFlexibleBottomMargin |
-                                          UIViewAutoresizingFlexibleRightMargin;
-  [self.alignmentButton addTarget:self
-                           action:@selector(changeAlignment:)
-                 forControlEvents:UIControlEventTouchUpInside];
-  [self.view addSubview:self.alignmentButton];
+  self.title = @"Tabs With Icons";
+}
+
+- (void)setupScrollView {
+  self.scrollView = [[UIScrollView alloc] initWithFrame:CGRectZero];
```

Comment:
This is called from viewDidLoad, not an init. So, let's leave it calling the setter. It's more conservative for public code.